src/creds.rs

```diff
@@ -185,6 +196,53 @@ impl Codec for BasicCredential {
     // }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct ExtensibleCredential {
```

Comment:
Can you put this new credential into its own module (file)? Then you can add doc comments there that describe how it works. Since it's a custom extension this needs a little more description.